### PR TITLE
Expand performance scaling to affect mining and crafting speeds

### PR DESCRIPTION
**Pull request by mheguy**

### CODE REVIEW

File: features/performance.lua [hunk 1]
```diff
@@ -4,56 +4,69 @@ local format = string.format
...
-    allowed_by_server = true,
-}, function (arguments, player)
-    local scale = tonumber(arguments.scale)
-    if scale == nil or scale < 0.05 or scale > 1 then
-        player.print('Scale must be a valid number ranging from 0.05 to 1')
-        return
+Command.add(
+    'performance-scale-set',
+    {
+        description = 'Sets the performance scale between 0.05 and 1. Will alter the game speed and character running speed per force.',
```
Comment (linaori) [inline features/performance.lua]:
```suggestion
        description = 'Sets the performance scale between 0.05 and 1. Will alter the game speed, manual mining speed, manual crafting speed and character running speed per force.',
```

File: features/performance.lua [hunk 2]
```diff
@@ -4,56 +4,69 @@ local format = string.format
...
+            player.print('Scale must be a valid number ranging from 0.05 to 1')
+            return
+        end
+
+        Performance.set_time_scale(scale)
+        local p = game.print
+        local stat_mod = Performance.get_player_stat_modifier()
+        p('## - Changed the game speed and running speed.')
+        p(format('## - Game speed: %.2f', Performance.get_time_scale()))
+        p(format('## - Force running speed: %.2f -- mining speed: %.2f -- crafting speed: %.2f', stat_mod, stat_mod, stat_mod))
```
Comment (linaori) [inline features/performance.lua]:
```suggestion
        p(format('## - Running speed: %.2f', stat_mod))
        p(format('## - Manual mining speed: %.2f', stat_mod))
        p(format('## - Manual crafting speed: %.2f', stat_mod))
```

File: features/performance.lua [hunk 3]
```diff
@@ -4,56 +4,69 @@ local format = string.format
...
+        local scale = tonumber(arguments.scale)
+        if scale == nil or scale < 0.05 or scale > 1 then
+            player.print('Scale must be a valid number ranging from 0.05 to 1')
+            return
+        end
+
+        Performance.set_time_scale(scale)
+        local p = game.print
+        local stat_mod = Performance.get_player_stat_modifier()
+        p('## - Changed the game speed and running speed.')
```
Comment (linaori) [inline features/performance.lua]:
```suggestion
        p('## - Game speed changed to compensate for UPS drops and players trying to catch up.')
```

**Review by linaori — CHANGES_REQUESTED** [review]
Nice changes! I have some minor text changes to add, using the github beta feature you should be able to apply the proposed changes directly if you agree with them

**Comment by mheguy** [comment]
That's pretty damn neat

**Review by linaori — APPROVED** [review]

**Pull request merged**